app/models/health_care_application.rb

```diff
@@ -272,9 +276,9 @@ def log_submission_failure_details
       'HCA total failure',
       :error,
       {
-        first_initial: parsed_form['veteranFullName']['first'][0],
-        middle_initial: parsed_form['veteranFullName']['middle'].try(:[], 0),
-        last_initial: parsed_form['veteranFullName']['last'][0]
+        first_initial: parsed_form.dig('veteranFullName', 'first')&.[](0) || 'no initial provided',
+        middle_initial: parsed_form.dig('veteranFullName', 'middle')&.[](0) || 'no initial provided',
+        last_initial: parsed_form.dig('veteranFullName', 'last')&.[](0) || 'no initial provided'
```

Comment:
I cherry picked these changes from https://github.com/department-of-veterans-affairs/vets-api/pull/17202 for this. I wanted to be able to test not providing the name here without it blowing up, and I will probably not merge that other PR as it has some failing specs in other related jobs that I'd like to better understand before I change it as I am not confident in the test coverage.